File: CatanServer/Catan.Application/Dtos/MapDto.cs
using Catan.Application.Dtos.GamePieces;
using Catan.Domain.Entities.GamePieces;

namespace Catan.Application.Dtos
{
    public class MapDto
	{
		public HexTileDto[] HexTiles { get; set; }
		public int ThiefPosition { get; set; }
		public List<SettlementDto> Settlements {get; set;}
		public List<CityDto> Cities { get; set; }
		public List<RoadDto> Roads { get; set; }
		public List<SpecialHarborDto> SpecialHarbors { get; set; }
		
	}
}

File: catan_app/src/interfaces/MapDto.ts
import {HexTileDto} from "./HexTileDto";
import {SettlementDto} from "./SettlementDto";
import {RoadDto} from "./RoadDto";
import {CityDto} from "./CityDto";
import {HarborDto} from "./HarborDto";

export interface MapDto {
    hexTiles: HexTileDto[];
    thiefPosition: number;
    settlements: SettlementDto[];
    cities: CityDto[];
    roads: RoadDto[];
    specialHarbors: HarborDto[];
}